govcd/vapp_test.go

```diff
@@ -853,6 +853,93 @@ func (vcd *TestVCD) Test_AddAndRemoveIsolatedVappNetwork(check *C) {
 	check.Assert(task.Task.Status, Equals, "success")
 }
 
+// Test_AddAndRemoveIsolatedVappNetworkIpv6 is identical to Test_AddAndRemoveIsolatedVappNetwork,
+// but it uses ipv6 values for network specification.
+func (vcd *TestVCD) Test_AddAndRemoveIsolatedVappNetworkIpv6(check *C) {
+	fmt.Printf("Running: %s\n", check.TestName())
+
+	vapp, err := deployVappForTest(vcd, "Test_AddAndRemoveIsolatedVappNetwork")
+	check.Assert(err, IsNil)
+	check.Assert(vapp, NotNil)
+
+	// Add Metadata
+	networkName := check.TestName()
+	description := "Created in test"
+	const gateway = "fe80:0:0:0:0:0:0:aaaa"
+	const prefixlength = "100"
+	// VCD API returns ipv6 addresses in expanded format, so this is
+	// needed to compare values properly.
+	const dns1 = "2001:4860:4860:0:0:0:0:8844"
+	const dns2 = "2001:4860:4860:0:0:0:0:8844"
+	const dnsSuffix = "biz.biz"
+	const startAddress = "fe80:0:0:0:0:0:0:aaab"
+	const endAddress = "fe80:0:0:0:0:0:0:bbbb"
+	const dhcpStartAddress = "fe80:0:0:0:0:0:0:cccc"
+	const dhcpEndAddress = "fe80:0:0:0:0:0:0:dddd"
+	const maxLeaseTime = 3500
+	const defaultLeaseTime = 2400
+	var guestVlanAllowed = true
+
+	vappNetworkSettings := &VappNetworkSettings{
+		Name:               networkName,
+		Gateway:            gateway,
+		SubnetPrefixLength: prefixlength,
+		DNS1:               dns1,
+		DNS2:               dns2,
+		DNSSuffix:          dnsSuffix,
+		StaticIPRanges:     []*types.IPRange{{StartAddress: startAddress, EndAddress: endAddress}},
+		DhcpSettings:       &DhcpSettings{IsEnabled: true, MaxLeaseTime: maxLeaseTime, DefaultLeaseTime: defaultLeaseTime, IPRange: &types.IPRange{StartAddress: dhcpStartAddress, EndAddress: dhcpEndAddress}},
+		GuestVLANAllowed:   &guestVlanAllowed,
+		Description:        description,
+	}
+
+	vappNetworkConfig, err := vapp.CreateVappNetwork(vappNetworkSettings, nil)
```

Comment:
Should we also check that
```
check.Assert(err, NotNil)
check.Assert(vappNetworkConfig, IsNil)
```
when we supply `NetMask` and `SubnetPrefixLength`?
Or in other words, add some checks to the non happy paths?
Or is it trivial hence not needed?

Reply:
I like this idea, adds test coverage too! Updated